mycroft/configuration/config.py

```diff
@@ -105,18 +106,30 @@ def load_local(self, path):
             except Exception as e:
                 LOG.error("Error loading configuration '{}'".format(path))
                 LOG.error(repr(e))
+                self.is_valid = False
         else:
             LOG.debug("Configuration '{}' not defined, skipping".format(path))
 
-    def store(self, path=None):
-        """Cache the received settings locally.
+    def store(self, path=None, force=False):
+        """Save config to disk.
 
         The cache will be used if the remote is unreachable to load settings
         that are as close to the user's as possible.
+
+        path (str): path to store file to, if missing will use the path from
+                    where the config was loaded.
+        force (bool): Set to True if writing should occur despite the original
+                      was malformed.
         """
         path = path or self.path
-        with open(path, 'w') as f:
-            json.dump(self, f, indent=2)
+        if self.is_valid or force:
+            # Only write settings
+            with open(path, 'w') as f:
+                json.dump(self, f, indent=2)
+        else:
+            LOG.warning(('"{}" was not a valid config file when loaded, '
+                         'will not save config. Please correct the json or '
+                         'remove it to allow updates.').format(path))
```

Comment:
Since this will rarely happen in the absence of user interaction, I think it would be helpful if Mycroft spoke.

Reply:
I can add a return value to the method so skills writing to the config can check if the write failed and perform the speak action.